core/chains/evm/gas/models.go

```diff
@@ -171,6 +171,33 @@ func (fee EvmFee) ValidDynamic() bool {
 	return fee.DynamicFeeCap != nil && fee.DynamicTipCap != nil
 }
 
+// Helper function to get the minimum of two values
+func min(a, b uint64) uint64 {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+// Utility function to calculate the highest gas price or fee for a given limit
+func getHighestPrice(maxGasCost, gasLimit uint64) uint64 {
+	return maxGasCost / gasLimit
+}
+
+// For legacy transactions
+func getLegacyGasPrice(calculatedGasPrice, maxGasCost, calculatedGasLimit uint64) *assets.Wei {
+	highestPrice := getHighestPrice(maxGasCost, calculatedGasLimit)
+	return assets.NewWei(new(big.Int).SetUint64(min(highestPrice, calculatedGasPrice)))
```

Comment:
Does `highest` mean something different than `max`? If not, then we should stick to just one term. IMHO this reads much more easily:
```suggestion
	maxPrice := maxGasCost / calculatedGasLimit
	return assets.NewWei(new(big.Int).SetUint64(min(maxPrice, calculatedGasPrice)))
```